### PR TITLE
Add Gridding Cookbook to the gallery

### PR DESCRIPTION
**Pull request by r-ford**
This cookbook was removed in #142

### CODE REVIEW

**Comment by r-ford** [comment]
It looks like the issue is still there.

**Comment by brian-rose** [comment]
After some local debugging, the reason turns out to be [this line in the _gallery_info.yml file](https://github.com/ProjectPythia/gridding-cookbook/blob/7426eb95e8bac20651c5c67d1d55bb25b01f2542/_gallery_info.yml#L3) which has an empty tag:
```
tags:
  domains: 
  packages: 
    - xarray
```
etc.

Turns out our code doesn't know how to handle an empty set of tags.

A simple workaround would be to remove the `domains` field from the `_gallery_info.yml` file in the gridding cookbook repo. But our code shouldn't be that fragile. I'll see if there's a straightforward fix in the gallery code that we can implement here.

**Comment by brian-rose** [comment]
This should work.

The gallery code already had a check for empty tags but there was a bug in it. It was checking to see if the first item in a list was `None`, but there is no list at all in this case, just a plain old `None`.

File: site/_extensions/gallery_generator.py [hunk 1]
```diff
@@ -68,7 +68,7 @@ def generate_repo_dicts(all_items):
             gallery_info_dict = yaml.safe_load(urllib.request.urlopen(gallery_info_url))
             thumbnail = gallery_info_dict["thumbnail"]
             tag_dict = {
-                k: v for k, v in gallery_info_dict["tags"].items() if v[0] is not None
+                k: v for k, v in gallery_info_dict["tags"].items() if v is not None
```
Comment (jukent) [inline site/_extensions/gallery_generator.py:71]: Do we want to check for both `v is not None` and `v[0] is not None`?

Comment (brian-rose) [reply site/_extensions/gallery_generator.py]: Can't hurt. I just implemented that in the latest push.

**Pull request merged**